things/service/src/main/java/org/eclipse/ditto/things/service/persistence/actors/strategies/commands/MigrateThingDefinitionStrategy.java

```diff
@@ -0,0 +1,279 @@
+ */
+package org.eclipse.ditto.things.service.persistence.actors.strategies.commands;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import org.apache.pekko.actor.ActorSystem;
+import org.eclipse.ditto.base.model.entity.metadata.Metadata;
+import org.eclipse.ditto.base.model.headers.DittoHeaders;
+import org.eclipse.ditto.base.model.headers.entitytag.EntityTag;
+import org.eclipse.ditto.base.model.json.FieldType;
+import org.eclipse.ditto.internal.utils.persistentactors.results.Result;
+import org.eclipse.ditto.json.JsonFactory;
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonObjectBuilder;
+import org.eclipse.ditto.json.JsonPointer;
+import org.eclipse.ditto.placeholders.PlaceholderFactory;
+import org.eclipse.ditto.placeholders.TimePlaceholder;
+import org.eclipse.ditto.policies.model.ResourceKey;
+import org.eclipse.ditto.rql.parser.RqlPredicateParser;
+import org.eclipse.ditto.rql.query.filter.QueryFilterCriteriaFactory;
+import org.eclipse.ditto.rql.query.things.ThingPredicateVisitor;
+import org.eclipse.ditto.things.model.Thing;
+import org.eclipse.ditto.things.model.ThingId;
+import org.eclipse.ditto.things.model.ThingsModelFactory;
+import org.eclipse.ditto.things.model.signals.commands.ThingCommandSizeValidator;
+import org.eclipse.ditto.things.model.signals.commands.ThingResourceMapper;
+import org.eclipse.ditto.things.model.signals.commands.modify.MergeThing;
+import org.eclipse.ditto.things.model.signals.commands.modify.MigrateThingDefinition;
+import org.eclipse.ditto.things.model.signals.events.ThingEvent;
+
+
+/**
+ * Strategy to handle the {@link MigrateThingDefinition} command.
+ *
+ * This strategy processes updates to a Thing's definition, applying necessary data migrations
+ * and ensuring that defaults are properly initialized when required.
+ *
+ * Assumptions:
+ * - The {@link MigrateThingDefinition} command provides a ThingDefinition URL, which is used
+ *   to create a skeleton Thing. The command's payload also includes migration data and patch
+ *   conditions for fine-grained updates.
+ * - Patch conditions are evaluated using RQL-based expressions to determine which migration
+ *   payload entries should be applied.
+ * - Skeleton generation extracts and merges Thing definitions and default values separately,
+ *   ensuring a clear distinction between structural updates and default settings.
+ * - After applying skeleton-based modifications and migration payloads, the changes are merged
+ * - Property initialization can be optionally enabled via the command, applying default values
+ *   to the updated Thing when set to true.
+ * - The resulting Thing undergoes validation to ensure compliance with WoT model constraints
+ *   before persisting changes.
```

Comment:
when doing a list in JavaDoc, you should use:
```
<ul>
<li>...</li>
</ul>
```
This will be properly rendered as list in e.g. IDE popups.